edgeless_telemetry/src/elasticsearch_resources.rs

```diff
@@ -0,0 +1,71 @@
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+// use sysinfo::System::SystemExt;
+use chrono::{DateTime, Utc};
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::Duration;
+use sysinfo::*;
+
+use crate::elasticsearch_api;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SystemResources {
+    cpu_percent: f32,
+    memory_percent: f32,
+    timestamp: DateTime<Utc>,
+}
+
+fn convert_to_value(data: &SystemResources) -> Value {
+    json!({
+        "cpu_percent": data.cpu_percent,
+        "memory_percent": data.memory_percent,
+        "timestamp": data.timestamp,
+    })
+}
+
+#[tokio::main]
```

Comment:
You can't use a main here. This is a library crate and the crates using it already have a main method.
You probably want this as a task that can be started from another service.

If you need a separate telemetry service, please add a dedicated bin (look at edgeless_node_d etc).